Framework/PythonInterface/mantid/plots/plotfunctions.py

```diff
@@ -204,18 +204,14 @@ def plot(workspaces, spectrum_nums=None, wksp_indices=None, errors=False,
 
     if not overplot:
         fig.canvas.set_window_title(figure_title(workspaces, fig.number))
-    else:
-        if ax.is_waterfall():
-            for i in range(len(nums)*len(workspaces)):
-                errorbar_cap_lines = datafunctions.remove_and_return_errorbar_cap_lines(ax)
-                datafunctions.convert_single_line_to_waterfall(ax, len(ax.get_lines()) - (i + 1))
-
-                if ax.waterfall_has_fill():
-                    datafunctions.waterfall_update_fill(ax)
-
-                ax.lines += errorbar_cap_lines
-
-    if superplot and not waterfall and not tiled:
+    elif ax.is_waterfall():
+        _overplot_waterfall(ax,len(nums)*len(workspaces))
+
+    if ax.is_waterfall():
```

Comment:
Thanks for making those changes. Just noticed we can probably remove this line, because we have the elif statement above it 
```suggestion
```

Reply:
We might also need the `elif superplot and not tiled` line to go back to `if superplot and not waterfall and not tiled`, but its worth checking if this is the case

Reply:
We still need to adjust the axis if we are not overplotting. If this `if` statement is removed and these lines brought under the scope of the preceding `elif` statement, that won't happen.

Reply:
Ah good point 👍